neo/rawio/intanbinaryrawio.py

```diff
@@ -0,0 +1,438 @@
+    stream_type_to_name,
+)
+
+
+class IntanBinaryRawIO(BaseRawIO):
+    """
+    Class for processing Intan Data when saved in binary format. Requires an
+    `info.rhd` as well one file per signal stream or one file per channel
+
+    Parameters
+    ----------
+    dirname: str, Path
+        The root directory containing the info.rhd file
+
+    """
+
+    extensions = ["rhd", "dat"]
+    rawmode = "one-dir"
+
+    def __init__(self, dirname=""):
+        BaseRawIO.__init__(self)
+        self.dirname = dirname
+
+    def _source_name(self):
+        return self.dirname
+
+    def _parse_header(self):
+
+        dir_path = Path(self.dirname)
+        assert dir_path.is_dir(), (
+            "IntanBinaryRawIO requires the root " "directory containing info.rhd"
+        )
+
+        header_file = dir_path / "info.rhd"
+
+        for file in possible_raw_files:
+            if (dir_path / file).is_file():
+                one_file_per_signal = True
+                break
+            else:
+                one_file_per_signal = False
+
+        self.one_file_per_signal = one_file_per_signal
+
+        if one_file_per_signal:
+            raw_file_dict = create_one_file_per_signal_dict(dir_path)
+        else:
+            raw_file_dict = create_one_file_per_channel_dict(dir_path)
+
+        (
+            self._global_info,
+            self._ordered_channels,
+            data_dtype,
+            self._block_size,
+        ) = read_rhd(header_file)
+
+        self._raw_data = {}
+        for stream_index, sub_datatype in data_dtype.items():
+            if one_file_per_signal:
+                self._raw_data[stream_index] = np.memmap(
+                    raw_file_dict[stream_index], dtype=sub_datatype, mode="r"
+                )
+            else:
+                self._raw_data[stream_index] = []
+                for channel_index, datatype in enumerate(sub_datatype):
+                    self._raw_data[stream_index].append(
+                        np.memmap(
+                            raw_file_dict[stream_index][channel_index],
+                            dtype=[datatype],
+                            mode="r",
+                        )
+                    )
+
+        # check timestamp continuity
+        if one_file_per_signal:
+            timestamp = self._raw_data[6]["timestamp"].flatten()
+        else:
+            timestamp = self._raw_data[6][0]["timestamp"].flatten()
+        assert np.all(np.diff(timestamp) == 1), "timestamp have gaps"
+
+        # signals
+        signal_channels = []
+        for c, chan_info in enumerate(self._ordered_channels):
+            name = chan_info["native_channel_name"]
+            chan_id = str(c)  # the chan_id have no meaning in intan
+            if chan_info["signal_type"] == 20:
+                # exception for temperature
+                sig_dtype = "int16"
+            else:
+                sig_dtype = "uint16"
+            stream_id = str(chan_info["signal_type"])
```

Comment:
Right now you are only getting numbers but we should use an enum or a simple type to make this meaningful as in the table:

![image](https://github.com/NeuralEnsemble/python-neo/assets/6429509/6d55bbb4-5d46-4865-8cc8-636626e2d5c7)

I also think yo can use those more meaningful string values as keys to your dictionaries as right now you have integers as keys which I think could be better.

Reply:
I agree, but in Neo you have to give `stream_index` which is an integer value mapping to signal_type for Intan. We have this in the IntanRawIO:
https://github.com/NeuralEnsemble/python-neo/blob/c96ec936049a125d6bedfa26b67514f927bb1ab3/neo/rawio/intanrawio.py#L436C1-L443C2.

But maybe you have a grander idea? We could do a deeper rewrite, but the user facing api still needs to accept an integer value.

Reply:
https://github.com/NeuralEnsemble/python-neo/blob/c96ec936049a125d6bedfa26b67514f927bb1ab3/neo/rawio/intanrawio.py#L110-L114

Each signal_stream has an `id` and a `name`. But for the most part we use the `stream_index`. We could rewrite this, but I was trying to maintain a similar feel to IntanRawIO in general.

Reply:
I think this is a good argument. Lets keep it similar to the other class and change later.